src/custom/connection/index.ts

```diff
@@ -27,6 +22,24 @@ export interface Connection {
   type: ConnectionType
 }
 
+/**
+ * To avoid including external libs for wallet connection in the bundle
+ * We load them in runtime by demand
+ */
+class AsyncConnector extends Connector {
+  constructor(private loader: () => Promise<Connector>, actions: Actions, onError?: (error: Error) => void) {
+    super(actions, onError)
+  }
+
+  activate(...args: unknown[]): Promise<void> | void {
+    return this.loader().then((connector) => {
+      // There is a magic - we change async-connector prototype to the loaded connector
+      ;(this as any).__proto__ = connector
+      return connector.activate(...args)
```

Comment:
Curious to how you got to this magic 🤔

Reply:
I very wanted to get rid of these packages from the bundle :)

Reply:
Also, it was required in Tinkoff to understand how prototype and prototypal inheritance works.
It's the first time in my life, when I have a chance to use this knowledge in practice.

Reply:
> I very wanted to get rid of these packages from the bundle :)

Yes, I got that part :)

> Also, it was required in Tinkoff to understand how prototype and prototypal inheritance works.
It's the first time in my life, when I have a chance to use this knowledge in practice.

That's what I wanted to know, how you got to this prototype magic ✨

Reply:
Hehe, before ES6 it was very important to understand the prototype. Before I use to do prototype questions in my interviews.

I also don't get how you arrived here :) 
so maybe is a good material for the Jam Session tomorrow too (to share with the team this little tricks).